src/requests/UsersApi.ts

```diff
@@ -0,0 +1,192 @@
+
+        const parameters: URLSearchParams = new URLSearchParams();
+
+        if (typeof userId !== 'string') {
+            throw new Error('No user id was provided!');
+        }
+        if (userId.length < 1) {
+            throw new Error('Empty User id or no User id were provided!');
+        }
+
+        parameters.append('userId', userId);
+
+        const paramRequest: VRCAPI.Generics.executeRequestType = {
+            currentRequest: ApiPaths.users.getUserGroups,
+            pathFormated: ApiPaths.users.getUserGroups.path.replace('{userId}', userId),
+            queryOptions: parameters,
+        };
+
+        this.checkValidData(paramRequest);
+        const queryResult = await this.executeRequest<VRCAPI.Groups.Models.Group[]>(paramRequest);
+
+        return queryResult;
+    }
+
+}
+
+    /**
+     * This function is used to get the rank tag of a user. Returns the highest rank tag of the user. Else returns undefined.
+     * @param user The User or CurrentUser object to get the rank tag from.
+     * @returns The rank tag of the user. If the user has no rank tag, returns undefined.
+     */
+    export function getVRCRankTag(user: VRCAPI.Users.Models.User | VRCAPI.Users.Models.CurrentUser): VRCAPI.Users.Models.VRCRanks | undefined {
+        // the highest vrcrank we can find in the user's tag is the rank of the user we should return
+        const tags: VRCAPI.Generics.AllTags[] = user.tags;
+        if (user.tags.includes(VRCRanks.Veteran)) {
+            return VRCRanks.Veteran;
+        } else if (tags.includes(VRCRanks.Trusted)) {
+            return VRCRanks.Trusted;
+        } else if (tags.includes(VRCRanks.Known)) {
+            return VRCRanks.Known;
+        } else if (tags.includes(VRCRanks.User)) {
+            return VRCRanks.User;
+        } else if (tags.includes(VRCRanks.New)) {
+            return VRCRanks.New;
+        } else if (!tags.includes(VRCRanks.Veteran) &&
+            !tags.includes(VRCRanks.Trusted) &&
+            !tags.includes(VRCRanks.Known) &&
+            !tags.includes(VRCRanks.User) &&
+            !tags.includes(VRCRanks.New) &&
+            !tags.includes(VRCRanks.Nuisance)) {
+
+            return VRCRanks.Visitor;
+        } else if (tags.includes(VRCRanks.Nuisance)) {
+            return VRCRanks.Nuisance;
+        } else {
+            return undefined;
+        }
+
+    }
+
+
+    /**
+     * Get the rank of the user as a string.
+     * @param user The User or Current User object to get the rank name from.
+     * @returns The rank name of the user. If the user has no rank tag, returns "Rank not found."
+     */
+    export function getVRCRankName(user: VRCAPI.Users.Models.User | VRCAPI.Users.Models.CurrentUser): string {
+        const rank = getVRCRankTag(user);
+        if (!rank) {
+            return "Rank not found.";
+        }
+        return VRCRanksName[rank];
+    }
```

Comment:
If you solve the problem of the getVRCRankTag function you will see that this here is useless.

Reply:
There is tags from vrchat like "system_trust_known", there is the enum Known and there is the definition "Known User". That's why I did it the way I did.